pennylane/pauli/dla/lie_closure.py

```diff
@@ -13,14 +13,147 @@
 # limitations under the License.
 """A function to compute the Lie closure of a set of operators"""
 # pylint: disable=too-many-arguments
+import itertools
 from functools import reduce
 
+from typing import Union, Iterable
 from copy import copy
 import numpy as np
 
 import pennylane as qml
 
 from pennylane.operation import Operator
+from ..pauli_arithmetic import PauliWord, PauliSentence
+
+
+def lie_closure(
```

Comment:
Could this also make sense as `lie_algebra`? i.e., it calculates the Lie algebra for you.

Reply:
Colloquially that could be understood, but I think its rather uncommon. "Lie closure" is, afaik, the most common and apt way to describe this process